UI.py

```diff
@@ -347,6 +365,41 @@ def FormatGetEndOfMonthReportEmbed(member, month, year):
         
     return member_data
 
+def CreateGetEndOfYearReportEmbed(members_list, year = None):
+    year = year or datetime.datetime.now().year
+    embed = discord.Embed(
+        title = f'End of Year Report',
+        description = f'{year} Report:',
+        colour = 0x4682B4
+    )
+    embed.set_thumbnail(url = os.getenv("Salary_Image"))
+    embed.add_field(name = '\u200B', value = '\u200B', inline = False)
+
+    for member in members_list:
+        member_data = FormatGetEndOfYearReportEmbed(member, year)
+        embed.add_field(name = f'**{member["name"].upper()}**', value = member_data, inline = False)
+        embed.add_field(name = '\u200B', value = '\u200B', inline = False)
+
+    footer_text = (("\u200B " * embed_footer_spaces_count) + datetime.date.today().strftime("%d/%m/%Y"))
+    embed.set_footer(text = footer_text)
+    return embed
+
+def FormatGetEndOfYearReportEmbed(member, year):
+    member_data = ""
+    start_date = datetime.datetime(year, 1, 1)
+    end_date = datetime.datetime(year, 12, utils.GetMonthDaysCount(12, year))
+    paid_leaves = leave_db.GetApprovedPaidLeaves(member["id"], start_date, end_date)
+    sick_leaves = leave_db.GetApprovedSickLeaves(member["id"], start_date, end_date)
+    emergency_leaves = leave_db.GetApprovedEmergencyLeaves(member["id"], start_date, end_date)
+    remaining_leaves_balance = leave_db.GetAnnualLeaveBalance(member["id"])
+    bonus_precentage = utils.CalculatePercentage(float(os.getenv("Average_Working_Days_Count")), remaining_leaves_balance)
```

Comment:
isn't this upside down? aren't we doing remaining leaves divided by average working days?

Reply:
avg working days are the total in this eq: 

20.5 => 100% salary
then
remaining leaves => ??%

then ??% = (remaining leaves * 100 ) / 20.5 it will give you the bonus percentage, or i did i miss understood you question ?

Reply:
if you have 5 leaves, you're going to get (5/20.5 * 100)% additional payment. 
5 first, then 20.5

if that function works the other way round, please change it

Reply:
Done